client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -92,49 +96,67 @@
-    private         TemporalUnit                                      timeoutUnit    = ChronoUnit.SECONDS;
+    private final   Map<Class<?>, List<HttpClient.BeanParamProperty>> beanParamCache   = new HashMap<>();
+    private final   Map<Method, JaxRsMeta>                            jaxRsMetaMap     = new ConcurrentHashMap<>();
+    private         int                                               timeout          = 10;
+    private         TemporalUnit                                      timeoutUnit      = ChronoUnit.SECONDS;
+    private         Set<String>                                       sensitiveHeaders = new TreeSet<>(Comparator.comparing(String::toLowerCase));
     private final   Duration                                          retryDuration;
 
     @Inject
     public HttpClient(HttpClientConfig config,
-                             ReactorRxClientProvider clientProvider,
-                             ObjectMapper objectMapper,
-                             RequestParameterSerializers requestParameterSerializers,
-                             Set<PreRequestHook> preRequestHooks
+                      ReactorRxClientProvider clientProvider,
+                      ObjectMapper objectMapper,
+                      RequestParameterSerializers requestParameterSerializers,
+                      Set<PreRequestHook> preRequestHooks
     ) {
-        this.config = config;
+        this.config         = config;
         this.clientProvider = clientProvider;
-        this.objectMapper = objectMapper;
+        this.objectMapper   = objectMapper;
         this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         this.requestParameterSerializers = requestParameterSerializers;
 
-        serverInfo = new InetSocketAddress(config.getHost(), config.getPort());
-        collector = new ByteBufCollector(config.getMaxResponseSize());
+        serverInfo           = new InetSocketAddress(config.getHost(), config.getPort());
+        collector            = new ByteBufCollector(config.getMaxResponseSize());
         this.preRequestHooks = preRequestHooks;
-        this.retryDuration = Duration.ofMillis(config.getRetryDelayMs());
+        this.retryDuration   = Duration.ofMillis(config.getRetryDelayMs());
     }
 
     public HttpClient(HttpClientConfig config) {
         this(config, new ReactorRxClientProvider(config, new HealthRecorder()), new ObjectMapper(), new RequestParameterSerializers(), emptySet());
     }
 
     public static void setTimeout(Object proxy, int timeout, ChronoUnit timeoutUnit) {
+       ifHttpClientDo(proxy, handler -> handler.setTimeout(timeout, timeoutUnit));
+    }
+
+    public static void markHeaderAsSensitive(Object proxy, String header) {
+        markHeadersAsSensitive(proxy, singleton(header));
+    }
+
+    public static void markHeadersAsSensitive(Object proxy, Set<String> headers) {
+        ifHttpClientDo(proxy, handler -> handler.setSensitiveHeaders(headers));
```

Comment:
Nitpicking here but maybe here and in setTimout name the lambda parameter httpClient to make it clearer that it is in fact an httpClient instance coming in.